src/sql/workbench/contrib/notebook/browser/models/notebookModel.ts

```diff
@@ -449,7 +449,9 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (!this._activeClientSession) {
 				this.updateActiveClientSession(clientSession);
 			}
-			let profile = new ConnectionProfile(this._notebookOptions.capabilitiesService, this.connectionProfile);
+
+			// If a connection profile is passed in and _activeConnection isn't yet set, use that. Otherwise, use _activeConnection
+			let profile = this._activeConnection ? this._activeConnection : new ConnectionProfile(this._notebookOptions.capabilitiesService, this.connectionProfile);
```

Comment:
So while looking at this I noticed something odd - on [line 683](https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/contrib/notebook/browser/models/notebookModel.ts#L683) you're setting _activeConnection to the new connection but then in the next line you're calling setActiveConnectionIfDifferent which is never going to do anything because the id's will always be the same.

Reply:
lemme take a look at that separately. i'd like to keep this change as minimal as possible and not introduce any other risk into this PR if at all possible.